src/app/shared/guards/auth.guard.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivateChild, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable, tap } from 'rxjs';
+import { AuthService } from '@shared/services/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivateChild {
+
+  constructor(
+    private authService: AuthService,
+    private router: Router
+  ) {
+  }
+  canActivateChild(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.authService.isAuthenticated()
+      .pipe(tap(res => {
```

Comment:
No need to use tap here. You can use map return UrlTree, which redirects

Reply:
fixed